modules/services/kanata.nix

```diff
@@ -0,0 +1,171 @@
+          description = ''
+            The config file.
+
+            By default, it is generated by values from other kanata
+            module options.
+
+            You can also set it to your own full config file which
+            overrides all other kanata module options.  ${upstreamDoc}
+          '';
+        };
+        extraArgs = lib.mkOption {
+          type = lib.types.listOf lib.types.str;
+          default = [ ];
+          description = "Extra command line arguments passed to kanata.";
+        };
+        port = lib.mkOption {
+          type = lib.types.nullOr lib.types.port;
+          default = null;
+          example = 6666;
+          description = ''
+            Port to run the TCP server on. `null` will not run the server.
+          '';
+        };
+      };
+    };
+
+  mkName = name: "kanata-${name}";
+
+  mkDevices =
+    devices:
+    let
+      devicesString = lib.pipe devices [
+        (map (device: "\"" + device + "\""))
+        (lib.concatStringsSep " ")
+      ];
+    in
+    lib.optionalString ((lib.length devices) > 0) "macos-dev-names-include (${devicesString})";
+
+  mkConfig =
+    name: keyboard:
+    pkgs.writeTextFile {
+      name = "${mkName name}-config.kdb";
+      text = ''
+        (defcfg
+          ${keyboard.extraDefCfg}
+          ${mkDevices keyboard.devices})
+
+        ${keyboard.config}
+      '';
+      # Only the config file generated by this module is checked.  A
+      # user-provided one is not checked because it may not be available
+      # at build time.  I think this is a good balance between module
+      # complexity and functionality.
+      checkPhase = ''
+        ${lib.getExe cfg.package} --cfg "$target" --check --debug
+      '';
+    };
+
+  mkService =
+    name: keyboard:
+    lib.nameValuePair (mkName name) {
+      command = ''
+        ${lib.getExe cfg.package} \
+           --cfg ${keyboard.configFile} \
+           ${lib.optionalString (keyboard.port != null) "--port ${toString keyboard.port} \\"}
+           ${lib.optionalString (keyboard.extraArgs != "") builtins.concatStringsSep " " keyboard.extraArgs}
+      '';
+      serviceConfig = {
+        Label = "org.nixos.kanata";
+        KeepAlive = true;
+        RunAtLoad = true;
+      };
+    };
+in
+{
+  options.services.kanata = {
+    enable = lib.mkEnableOption "kanata, a tool to improve keyboard comfort and usability with advanced customization";
+    package = lib.mkPackageOption pkgs "kanata" {
+      example = [ "kanata-with-cmd" ];
+      extraDescription = ''
+        ::: {.note}
+        If {option}`danger-enable-cmd` is enabled in any of the keyboards, the
+        `kanata-with-cmd` package should be used.
+        :::
+      '';
+    };
+    keyboards = lib.mkOption {
+      type = lib.types.attrsOf (lib.types.submodule keyboard);
+      default = { };
+      description = "Keyboard configurations.";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
```

Comment:
theres no enabling of the driverkit which is required

Reply:
my pr #1595 adds a separate module to enable this which should probably be enabled here and maybe with the src overriden to be 5.0.0 however it it will soon be compatible with 6.2.0